docs/modules/ROOT/pages/explanations/decisions/forgejo.adoc

```diff
@@ -0,0 +1,346 @@
+= Forgejo / Codey
+
+This page discusses the technical details for our two products https://products.vshn.ch/appcat/forgejo.html[Forgejo by VSHN^] and https://products.vshn.ch/codey.html[Codey^]: Fully managed, dedicated https://forgejo.org/[Forgejo^] instances.
+
+Codey is a product name for a very specific configuration and product offering of Forgejo by VSHN.
+Codey won't be available on the regular in-cluster VSHN AppCat offering, only Forgejo by VSHN will.
+An instance of Codey has to be ordered via the Codey website and is running on pre-defined "black-box" OpenShift OKE service clusters to which the customer doesn't have access to.
+
+The offering of Codey combines compute resources, storage and the managed service into a pre-defined package.
+
+Main differences between Forgejo by VSHN (the AppCat service) and Codey:
+
+[cols=",,",options="header"]
+|===
+|
+|Forgejo by VSHN
+|Codey
+
+h|Compute and Storage
+|not included
+|included
+
+h|Resource Restrictions
+|depending on underlying infrastructure
+|constraint be the product plan
+
+h|Instance Location
+|depends on the underlying cluster
+|pre-defined by Codey
+
+h|Configuration Flexibility
+|most options can be configured
+|only few options are exposed
+|===
+
+To begin with, we'll use the simplest possible configuration, not using any external services for database, caches, etc.
+This also means there is no HA available.
+
+== Restrictions of Forgejo Features
+
+Due to technical limitations, we cannot support all possible features of Forgejo.
+This section lists these restrictions.
+
+Git access::
+Git access is only possible via HTTPS, SSH is not supported.
+Supporting SSH would require to expose a TCP service which inclines the use of a `LoadBalancer` type service.
+While not impossible, it complicates the configuration by a lot.
+
+Forgejo Actions::
+https://forgejo.org/docs/latest/admin/actions/[Forgejo Actions^] are fully supported, but the https://forgejo.org/docs/latest/admin/runner-installation/[Actions runner^] need to be provided by the user right now.
+We plan to add a managed runner in the future as well.
+
+Exposed configuration options::
+Many features of Forgejo are configured in the `app.ini` file and not via the web interface.
+We expose https://forgejo.org/docs/latest/admin/config-cheat-sheet/[configuration options^] through the <<Deployment>> configuration.
+
+== Deployment
+
+Deployment and configuration of Forgejo happens with the official https://code.forgejo.org/forgejo-helm/forgejo-helm[upstream Helm Chart^] and the https://github.com/crossplane-contrib/provider-helm[Crossplane Provider for Helm^].
+In the Crossplane Composite Resource Definition (XRD) we expose all required parameters.
+
+We differentiate between Forgejo by VSHN and Codey, exposing a different XRD depending on the service.
```

Comment:
The way I understand it, is that `Codey` is just an opinionated deployment of a `VSHNForgejo` instance.

We could probably just think of the `Codey` CRD as a template for `VSHNForgejo`.

If someone creates a `Codey` claim we just create a `XVSHNForgejo` (the composite, not another claim!) with hardcoded settings. Could even just be a PnT composition TBH.